lib/fastlane/plugin/emerge/helper/github.rb

```diff
@@ -43,10 +43,7 @@ def self.previous_sha
         if is_push?
           github_event_data.dig(:before)
         else
-          shell_command = "git rev-parse HEAD^"
-          UI.command(shell_command)
-          stdout, _, status = Open3.capture3(shell_command)
-          stdout.strip if status.success?
+          github_event_data.dig(:before) || self.base_sha
```

Comment:
Any docs on what "before" is? We've seen a lot of times where what GitHub uses for these values is not very intuitive and it's better to just use git. I didn't see this one documented here: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

Reply:
I agree it's not intuitive hence why I wanted to avoid it.. but my git code doesn't appear to work in the Github CI context. I tested it yesterday and couldn't get it to work, otherwise I wouldn't have switched to this

my understanding is that `before` is essentially the `sha` from the last update to the PR. So if you push a second commit, the `before` value will be the first commit's sha which is what we want.

I did find a frustrating corner case where if I squashed my commits `before` would "hold onto" that old no longer accurate value... but given that I couldn't find a way to get git commands to work in the CI context, I don't see an alternative